### PR TITLE
Add condition on rendering medium connecting icon

### LINKED ISSUE #19: Make integration of Medium URL optional

### PR DESCRIPTION
**Pull request by ayu023ban**
### Description

Fixes #19  issue
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires migration.
- [ ] My change requires `npm install`.

### CODE REVIEW

File: src/components/blog/blog-page.js [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
+              <Icon
+                name="medium"
+                size="large"
+                link={true}
+                onClick={() =>
+                  window.open(`${MEDIUM_URL}${MEDIUM_PUBLICATION}`, "_blank")
+                }
+              />
+            </Segment>
+          ) : null}
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:38]: You can use `&&` here instead of a ternary operator.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
@ayu023ban 
You will also have to write a message on the blog section that "This group doesn't have any blogs as of now. Check back later." or something similar on these grounds.

While you're at it, make sure you consider the following suggestions:
- Revert each file's permission
- Remove `;` everywhere.
- The strings should be enclosed in `''`. However, JSX attributes' values are enclosed in `""`.
- The commit messages should start with the present tense. For example: Add condition on rendering medium connecting icon

File: src/components/blog/blog-page.js (hunk 1 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:38]: "Revert each file's permission" what does it means?

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (algomaster99) [inline .prettierrc:1]: Don't push this. We plan to integrate such things to `omniport-frontend` for consistency across the entire ecosystem.

File: src/components/blog/blog-page.js [hunk 3]
```diff
@@ -15,26 +15,41 @@ class Blog extends Component {
...
+
+        return (
+          <Container styleName="common.margin">
+            <Card.Group itemsPerRow={3} stackable doubling>
+              {this.props.apiBlogData.data.map((info, id) => (
+                <BlogDetail info={info} key={id} />
+              ))}
+            </Card.Group>
+            <Segment basic padded textAlign="center">
+              {MEDIUM_PUBLICATION && MEDIUM_PUBLICATION !== '' ? (
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:29]: In my opinion, if the data in the blog list is there, `MEDIUM_PUBLICATION` **won't** be an empty string. Maybe we can get rid of this condition.

File: src/components/blog/blog-page.js [hunk 4]
```diff
@@ -15,26 +15,41 @@ class Blog extends Component {
...
-              size="large"
-              link={true}
-              onClick={() =>
-                window.open(`${MEDIUM_URL}${MEDIUM_PUBLICATION}`, '_blank')
-              }
-            />
-          </Segment>
-        </Container>
-      )
+      if (this.props.apiBlogData.data && this.props.apiBlogData.data.length) {
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:18]: Just check `this.props.apiBlogData.data.length > 0`.

File: src/components/blog/blog-page.js [hunk 5]
```diff
@@ -15,26 +15,41 @@ class Blog extends Component {
...
+                `This group doesn't have any blogs as of now. Check back later.`
+              )}
+            </Segment>
+          </Container>
+        )
+      } else {
+        return (
+          <Container styleName="common.margin">
+            <Segment basic padded textAlign="center">
+              {`This group doesn't have any blogs as of now. Check back later.`}
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:48]: Need not be enclosed in `{` and `}`.

**Review by algomaster99 — CHANGES_REQUESTED** [review]

File: src/components/blog/blog-page.js (hunk 4 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:18]: but if in case the apiBlogData.data is undefined then the whole code will break. So that's why I added the extra condition for that

File: src/components/blog/blog-page.js (hunk 3 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:29]: In the backend the field is Blank = True. So technically it can be an empty string.

File: .prettierrc (hunk 2 above)
Comment (ayu023ban) [reply .prettierrc:1]: ok

File: src/components/blog/blog-page.js (hunk 5 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:48]: ok

File: src/components/blog/blog-page.js (hunk 3 above)
Comment (algomaster99) [reply src/components/blog/blog-page.js:29]: But if it's `""`, `this.props.apiBlogData.data.length` will be equal to `0`, right?

Comment (ayu023ban) [reply src/components/blog/blog-page.js:29]: yes

Comment (algomaster99) [reply src/components/blog/blog-page.js:29]: So it won't enter this condition anyway

File: src/components/blog/blog-page.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
-import React, { Component } from 'react'
-import { Card, Container, Segment, Icon, Loader } from 'semantic-ui-react'
+import React, { Component } from "react"
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:1]: @ayu023ban Be careful about such formatting changes

Comment (ayu023ban) [reply src/components/blog/blog-page.js:1]: this is happening due to the removal of .preetierrc configuration file. as whenever i save the editor automatically formats according to default settings.
what is the solution of this?

Comment (ayu023ban) [reply src/components/blog/blog-page.js:1]: as if we moved the config file to top level directory then we have load whole project in order for the config file to work

Comment (algomaster99) [reply src/components/blog/blog-page.js:1]: Disable prettier "workspace" configuration. And apply it to only those projects where you need them.

Comment (ayu023ban) [reply src/components/blog/blog-page.js]: ok

File: src/components/blog/blog-page.js [hunk 7]
```diff
@@ -5,7 +5,6 @@ import BlogDetail from './blog-post-card'
 import { urlApiBlog } from '../../urls'
 import { MEDIUM_URL } from '../../consts'
 
-import common from '../../css/page-common-styles.css'
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:8]: Why did you remove this CSS import?

File: src/components/blog/blog-page.js [hunk 8]
```diff
@@ -15,26 +14,37 @@ class Blog extends Component {
...
+                <BlogDetail info={info} key={id} />
+              ))}
+            </Card.Group>
+            <Segment basic padded textAlign="center">
+              <Icon
+                name="medium"
+                size="large"
+                link={true}
+                onClick={() =>
+                  window.open(`${MEDIUM_URL}${MEDIUM_PUBLICATION}`, "_blank")
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:33]: `'_blank'`

**Review by algomaster99 — CHANGES_REQUESTED** [review]

File: src/components/blog/blog-page.js (hunk 7 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:8]: it was not being anywhere in the file

File: src/components/blog/blog-page.js (hunk 8 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:33]: but it is in jsx element so shouldn't it be double quotes?

File: src/components/blog/blog-page.js (hunk 7 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js:8]: now i understand... forgot that common was in the string. will add it now

File: src/components/blog/blog-page.js (hunk 8 above)
Comment (algomaster99) [reply src/components/blog/blog-page.js]: As soon as you enter `{ ... }`, you start writing JS.

Comment (ayu023ban) [reply src/components/blog/blog-page.js]: Ok.
Added changes

File: src/components/blog/blog-page.js [hunk 9]
```diff
@@ -4,7 +4,6 @@ import { Card, Container, Segment, Icon, Loader } from 'semantic-ui-react'
 import BlogDetail from './blog-post-card'
 import { urlApiBlog } from '../../urls'
 import { MEDIUM_URL } from '../../consts'
-
 import common from '../../css/page-common-styles.css'
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:7]: Don't remove this line

File: src/components/blog/blog-page.js [hunk 10]
```diff
@@ -4,7 +4,6 @@ import { Card, Container, Segment, Icon, Loader } from 'semantic-ui-react'
 import BlogDetail from './blog-post-card'
 import { urlApiBlog } from '../../urls'
 import { MEDIUM_URL } from '../../consts'
-
```
Comment (algomaster99) [inline src/components/blog/blog-page.js:7]: Don't remove this line

**Review by algomaster99 — CHANGES_REQUESTED** [review]

File: src/components/blog/blog-page.js (hunk 10 above)
Comment (ayu023ban) [reply src/components/blog/blog-page.js]: done

**Review by algomaster99 — APPROVED** [review]
@ayu023ban great work :100:
